File: 1041-available-captures-for-rook/1041-available-captures-for-rook.py
class Solution:
    def numRookCaptures(self, board: List[List[str]]) -> int:
        rook_pos = [-1, -1]
        for i in range(8):
            for j in range(8):
                if board[i][j] == "R":
                    rook_pos = [i, j]
                    break
        count = 0
        # check up
        for i in range(rook_pos[0] - 1, -1, -1):
            if board[i][rook_pos[1]] == "p":
                count += 1
                break
            elif board[i][rook_pos[1]] == "B":
                break
        # check down
        for i in range(rook_pos[0] + 1, 8):
            if board[i][rook_pos[1]] == "p":
                count += 1
                break
            elif board[i][rook_pos[1]] == "B":
                break
        # check left
        for i in range(rook_pos[1] - 1, -1, -1):
            if board[rook_pos[0]][i] == "p":
                count += 1
                break
            elif board[rook_pos[0]][i] == "B":
                break
        # check right
        for i in range(rook_pos[1] + 1, 8):
            if board[rook_pos[0]][i] == "p":
                count += 1
                break
            elif board[rook_pos[0]][i] == "B":
                break
        return count

File: 1041-available-captures-for-rook/1041-available-captures-for-rook.java
class Solution {
    public int numRookCaptures(char[][] board) {
        int[] rookPos = {-1, -1};
        for(int i = 0; i < 8; i++) {
            for(int j = 0; j < 8; j++) {
                if (board[i][j] == 'R') {
                    rookPos[0] = i;
                    rookPos[1] = j;
                }
            }
        }
        int count = 0;
        // check up
        for(int i = rookPos[0] - 1; i >= 0; i--){
            if(board[i][rookPos[1]] == 'p') {
                count++;
                break;
            } else if(board[i][rookPos[1]] == 'B') {
                break;
            }
        }
        // check down
        for(int i = rookPos[0] + 1; i < 8; i++) {
            if(board[i][rookPos[1]] == 'p') {
                count++;
                break;
            } else if(board[i][rookPos[1]] == 'B') {
                break;
            }
        }
        // check left
        for(int i = rookPos[1] -1; i >= 0; i--) {
            if(board[rookPos[0]][i] == 'p'){
                count++;
                break;
            } else if(board[rookPos[0]][i] == 'B') {
                break;
            }
        }
        // check right
        for(int i = rookPos[1] + 1; i < 8; i++) {
            if(board[rookPos[0]][i] == 'p'){
                count++;
                break;
            } else if(board[rookPos[0]][i] == 'B') {
                break;
            }
        }
        return count;
    }
}